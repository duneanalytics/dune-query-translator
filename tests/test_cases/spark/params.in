select
    explode(
        sequence(
            to_date('{{1 - Start Date}}'),
            least(NOW(), to_date('{{2 - End Date}}')),
            interval 1 {{3 - Time Granularity}}
        )
    ) as time
