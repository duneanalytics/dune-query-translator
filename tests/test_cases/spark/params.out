/* Success! If you're still running into issues, check out https://dune.com/docs/query/syntax-differences/ or reach out in the #dune-sql Discord channel. */

SELECT
  time
FROM 
CROSS JOIN UNNEST(SEQUENCE(
  CAST(SUBSTR(CAST(CAST('{{1 - Start Date}}' AS TIMESTAMP) AS VARCHAR), 1, 10) AS DATE),
  LEAST(
    CURRENT_TIMESTAMP,
    CAST(SUBSTR(CAST(CAST('{{2 - End Date}}' AS TIMESTAMP) AS VARCHAR), 1, 10) AS DATE)
  ),
  INTERVAL '1' {{3 - Time Granularity}}
) /* WARNING: Check out the docs for example of time series generation: https://dune.com/docs/query/syntax-differences/ */) AS array_column(time)